bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/ValueFactory.java

```diff
@@ -52,11 +53,8 @@ public static Value createValueState(ChannelConfig config, String channelTypeID)
             case MqttBindingConstants.DIMMER:
                 value = new PercentageValue(config.min, config.max, config.step, config.on, config.off);
                 break;
-            case MqttBindingConstants.COLOR_RGB:
-                value = new ColorValue(true, config.on, config.off, config.onBrightness);
-                break;
-            case MqttBindingConstants.COLOR_HSB:
-                value = new ColorValue(false, config.on, config.off, config.onBrightness);
+            case MqttBindingConstants.COLOR:
+                value = new ColorValue(ColorMode.valueOf(config.colorMode), config.on, config.off, config.onBrightness);
```

Comment:
Does `ColorMode.valueOf` work without `config.colorMode` being converted to uppercase?

Reply:
Good point, thanks! I have changed the colorMode parameter values and made them uppercase :wink: